src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -9088,6 +9088,110 @@ .locals init (C.Awaiter V_0)
                 """);
         }
 
+        [Fact]
+        public void CustomAwaitable_InSwitch_SequencePoints()
+        {
+            var code = """
+                using System;
+                using System.Threading.Tasks;
+
+                class C
+                {
+                    static async Task Main()
+                    {
+                        var result = (await new C()) switch
```

Comment:
Would it also be worthwhile to have an `await` inside a `switch` expression arm?